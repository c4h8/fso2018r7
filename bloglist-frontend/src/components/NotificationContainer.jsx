import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

const NotificationContainer = ({ notifications }) => (
  <div>
    {notifications.map(notification =>
      <div key={notification.id} style={notification.style}>
        {notification.message}
      </div>
    )}
  </div>
);

NotificationContainer.propTypes = ({
  notifications: PropTypes.arrayOf(Object)
});

const mapStateToProps = state => ({
  notifications: state.notifications
});

export default connect(mapStateToProps)(NotificationContainer);
