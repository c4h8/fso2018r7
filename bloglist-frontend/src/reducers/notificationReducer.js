import * as types from '../actionTypes';

function reducer(state = [], action) {
  switch (action.type) {
  case types.DELETE_NOTIFICATION:
    return state.filter(n => n.id !== action.id);

  default:
    return state;

  }
}

export default reducer;
