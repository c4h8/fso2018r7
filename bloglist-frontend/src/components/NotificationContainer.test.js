import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux'
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import notificationReducer from '../reducers/notificationReducer';
import NotificationContainer from './NotificationContainer';


const mockState = {
  notifications: [
    {message: 'lol', id: 1}
  ]
}

const reducer = combineReducers({
  notifications: notificationReducer,
});

const store = createStore(
  reducer,
  mockState,
  applyMiddleware(thunk)
);


describe('NotificationContainer', () => {
  it('renders blogs', () => {
    const component = renderer.create(
      <Provider store={store}>
        <NotificationContainer />
      </Provider>
    );
    
    console.log(component.toJSON());
    expect(component.toJSON()).toMatchSnapshot();
  });
});
