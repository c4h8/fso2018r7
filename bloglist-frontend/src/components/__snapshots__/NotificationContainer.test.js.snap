// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationContainer renders blogs 1`] = `
<div>
  <div>
    lol
  </div>
</div>
`;
